sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncStoredProcedureResponse.java

```diff
@@ -8,24 +8,32 @@
 
 public class CosmosAsyncStoredProcedureResponse extends CosmosResponse<CosmosStoredProcedureProperties> {
 
-    private CosmosAsyncStoredProcedure storedProcedure;
-    private StoredProcedureResponse storedProcedureResponse;
+    private final CosmosAsyncStoredProcedure storedProcedure;
+    private final StoredProcedureResponse storedProcedureResponse;
 
-    CosmosAsyncStoredProcedureResponse(ResourceResponse<StoredProcedure> response, CosmosAsyncContainer cosmosContainer) {
+    CosmosAsyncStoredProcedureResponse(ResourceResponse<StoredProcedure> response,
+                                       CosmosAsyncContainer cosmosContainer) {
         super(response);
-        if(response.getResource() != null){
+        if (response.getResource() != null) {
             super.setProperties(new CosmosStoredProcedureProperties(response));
             storedProcedure = new CosmosAsyncStoredProcedure(this.getProperties().getId(), cosmosContainer);
+            storedProcedureResponse = null;
+        } else {
+            storedProcedureResponse = null;
```

Comment:
Can be done outside if else, duplicate statement in both if and else.

Reply:
Done